server/command.go

```diff
@@ -135,174 +305,12 @@ func (p *Plugin) ExecuteCommand(c *plugin.Context, args *model.CommandArgs) (*mo
-			mlog.Error(err.Error())
-			p.postCommandResponse(args, "Encountered an error trying to unsubscribe. Please try again.")
-			return &model.CommandResponse{}, nil
-		}
-
-		p.postCommandResponse(args, fmt.Sprintf("Succesfully unsubscribed from %s.", repo))
-		return &model.CommandResponse{}, nil
-	case "disconnect":
-		p.disconnectGitHubAccount(args.UserId)
-		p.postCommandResponse(args, "Disconnected your GitHub account.")
-		return &model.CommandResponse{}, nil
-	case "todo":
-		text, err := p.GetToDo(ctx, info.GitHubUsername, githubClient)
-		if err != nil {
-			mlog.Error(err.Error())
-			p.postCommandResponse(args, "Encountered an error getting your to do items.")
-			return &model.CommandResponse{}, nil
-		}
-		p.postCommandResponse(args, text)
-		return &model.CommandResponse{}, nil
-	case "me":
-		gitUser, _, err := githubClient.Users.Get(ctx, "")
-		if err != nil {
-			p.postCommandResponse(args, "Encountered an error getting your GitHub profile.")
-			return &model.CommandResponse{}, nil
-		}
-
-		text := fmt.Sprintf("You are connected to GitHub as:\n# [![image](%s =40x40)](%s) [%s](%s)", gitUser.GetAvatarURL(), gitUser.GetHTMLURL(), gitUser.GetLogin(), gitUser.GetHTMLURL())
-		p.postCommandResponse(args, text)
-		return &model.CommandResponse{}, nil
-	case "help":
-		text := "###### Mattermost GitHub Plugin - Slash Command Help\n" + strings.Replace(COMMAND_HELP, "|", "`", -1)
-		p.postCommandResponse(args, text)
-		return &model.CommandResponse{}, nil
-	case "":
-		text := "###### Mattermost GitHub Plugin - Slash Command Help\n" + strings.Replace(COMMAND_HELP, "|", "`", -1)
-		p.postCommandResponse(args, text)
-		return &model.CommandResponse{}, nil
-	case "settings":
-		if len(parameters) < 2 {
-			p.postCommandResponse(args, "Please specify both a setting and value. Use `/github help` for more usage information.")
-			return &model.CommandResponse{}, nil
-		}
-
-		setting := parameters[0]
-		if setting != SETTING_NOTIFICATIONS && setting != SETTING_REMINDERS {
-			p.postCommandResponse(args, "Unknown setting.")
-			return &model.CommandResponse{}, nil
-		}
-
-		strValue := parameters[1]
-		value := false
-		if strValue == SETTING_ON {
-			value = true
-		} else if strValue != SETTING_OFF {
-			p.postCommandResponse(args, "Invalid value. Accepted values are: \"on\" or \"off\".")
-			return &model.CommandResponse{}, nil
-		}
-
-		if setting == SETTING_NOTIFICATIONS {
-			if value {
-				p.storeGitHubToUserIDMapping(info.GitHubUsername, info.UserID)
-			} else {
-				p.API.KVDelete(info.GitHubUsername + GITHUB_USERNAME_KEY)
-			}
-
-			info.Settings.Notifications = value
-		} else if setting == SETTING_REMINDERS {
-			info.Settings.DailyReminder = value
-		}
-
-		p.storeGitHubUserInfo(info)
-
-		p.postCommandResponse(args, "Settings updated.")
-		return &model.CommandResponse{}, nil
+	if f, ok := p.CommandHandlers[command]; ok {
+		message, err := f(c, args, parameters, info)
```

Comment:
Is there any place where `err != nil`? If not, can we drop the error return value.

Reply:
`err` is always nil. I can remove it